_posts/2021-02-26-feather-nrf52840.md

```diff
@@ -0,0 +1,462 @@
+transistors in a given piece of computer hardware approximately doubles. In
+fact, the efforts to save cost and energy for IoT hardware are directly opposed
+to Moore's Law: every additional transistor, be it for calculation or storage,
+costs more money and leads to more energy usage. Given, that for the IoT a
+multiple of 10 of devices per person is expected, meaning at least 100 billion
+devices deployed, low cost and low energy usage is vital.
+
+The main code base of RIOT is written in the [C programming language][C], but
+bindings for [C++], [Rust], [Python], [JavaScript], and others exist as well.
+
+For RIOT, a board is defined as a module that defines a [microcontroller] and
+configuration parameters how this microcontroller can interact with other
+devices on this platform and the outside world. Colloquially speaking, a board
+is somewhat like a computer system (e.g. a PC), and the microcontroller the
+central processor (CPU).
+So the process of porting a board is providing these configuration parameters.
+
+Personally, I am more involved with the networking architecture of RIOT—I make
+sure RIOT can talk to the outside world. So my closest contact to hardware in
+that area is mostly the interaction with the network device. Thus, there was
+just never the need for me to port a board for RIOT. So why did I do it now?
+Mainly, because I wanted to try and learn and because I had this piece of
+hardware lying around, for which I knew, that the port would be relatively easy,
+considering my skill set.
+
+The board I ported is the [Adafruit Feather nRF52840]. I bought this board
+earlier this year mainly to save some shipping costs, but also because I like
+the form factor of Adafruit's Feather line and wanted to see more support for it
+in RIOT.  It then became incidentally part of the most current RIOT release
+[2020.01].
+
+I am intentionally leaving the process of [creating a pull request] out
+as I want to focus this article mainly on how to provide a board port, not how
+to get it upstream.
+
+# Creating the board port
+As I said, a board in RIOT is mostly just configuration, both for the build
+system and the actual code. But first I needed a place to put this
+configuration.
+For simplicity I cut out the “Adafruit” out of the board's identifying name—and
+because the [Adafruit Feather M0] already was called `feather-m0`. Hence, for
+RIOT's purpuses the board is now just called `feather-nrf52840`. Under this name
+I created a directory for the board configuration to reside in:
+[`boards/feather-nrf52840`][directory].
+
+There I first created a Makefile to tell the build system [`make`][make] that
+the directory is indeed describing a board. This file is called
+[`boards/feather-nrf52840/Makefile`][Makefile]:
+
+{% highlight Makefile %}
+MODULE = board
+
+include $(RIOTBASE)/Makefile.base
+{% endhighlight %}
+
+The first line tells the build system, that this is a module called `board` (all
```

Comment:
usually, there is no comma before "that": system that

Reply:
I know ;-) Appearently, I wasn't that used to _that_ a year ago.